### ISSUE TITLE
Information about git repository hash when ece2cmor3 is installed in develop mode

### ISSUE THREAD

**Issue by jmrgonza** [EC-Earth/ece2cmor3#724]
I found these attributes in the header of some nc files that I produced with ece2cmor3 v1.8.1:
```
                :ece2cmor_git_revision = "local unknown branch" ;
                ...
                :history = "2022-01-21T10:18:31Z ; CMOR rewrote data to be consistent with CMIP6, CF-1.7 CMIP-6.2 and CF standards.;\n",
                        "processed by ece2cmor v1.8.1, git rev. local unknown branch\n",
                        "" ;
```
Instead of "local unknow branch" there should be a hash of the commit in the ece2cmor3 repository that I am using to process my data. I installed ece2cmor3 in a conda environment using the "python setput.py develop" command (as a workaround to solve another unrelated issue).

I have been looking into the code, and I have found out that, during the standard package installation with `python setup.py install`, the ece2cmor3.__version__.py file that appears in the package installation directory is not exactly equal to the one in the github repository. During the installation, the hash of the last commit in the active branch is computed and added to the __version__.py as a variable named `sha`. When the package is installed with `python setup.py develop` none of this happens and the `sha` variable does not exists.

Without the `__version__.sha` variable, what happens during the `ece2cmor` script execution is that there is a call to the `git_get_hash` function in https://github.com/EC-Earth/ece2cmor3/blob/d0f73f68974a54b3c361dc95ef0a6e7d840c384f/ece2cmor3/cmor_utils.py#L44-L59
However, in this case the result that you get from this call depends on the directory from where you execute the `ece2cmor` script. In my case, I was executing it from my home directory (not a git repository) and got the "unknown local branch" result that appeared in the attributes of the nc files. But it could have been worse: if you execute the `ece2cmor` script from a directory where you have another clone of the ece2cmor3 project, where you have done a checkout of a different branch, the information returned by the `git_get_hash` function would be the commit id in the active branch in the current directory when you execute the `ece2cmor` script, not the version that you installed with `python setup.py develop` and you are actually running.

To avoid inconsistencies like this, I would suggest to fix the `git_get_hash` function to ensure that it is getting the git commit in the directory containing the code that you are running (I think it may be possible to get it from the `ece2cmor3.__file__` variable or using the `inspect` module). I would also recommend to add some command line option meaning "I know what I am doing", and stop the execution of the `ece2cmor` command if the `ece2cmor3.__version__.sha` variable is not defined and the "I know what I am doing" flag is not set.

**Comment by treerink** [comment]
I have added the `checkversion` branch with a little program `versioncheck`, which can be called like:
```
versioncheck -V
```
In the script you can change the `case` variable to test a few cases. See the different behaviour if one uses:
```
python setup.py install
```
or
```
python setup.py develop
```

**Comment by treerink** [comment]
The console call is fixed in 227d15891f34e5b814b687c179677daabfc2cc54:
```
versioncheck --version
```
```
cd ${HOME}/cmorize/ece2cmor3/ece2cmor3/scripts/
./versioncheck.py --version
```

Further in both `genecec` as in `ece2cmor3` `log.info` the ece2cmor3 version & branch and the short & long ece2cmor_git_version hash has been added in 86a418b7abe903b30523603014c84c0bed62a93d & ae2b16562e1186a513f1300126ffe62d1542c71f
